grpc/core/src/main/java/io/helidon/grpc/core/model/Instance.java

```diff
@@ -0,0 +1,143 @@
+/*
+ * Copyright (c) 2019 Oracle and/or its affiliates. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.helidon.grpc.core.model;
+
+import java.util.Objects;
+import java.util.function.Supplier;
+
+import io.grpc.Status;
+
+/**
+ * A supplier of instances of objects.
+ */
+public interface Instance {
```

Comment:
This too seems a bit generic, and I question how relevant it will be once we start using CDI to create and manage instances?